src/utils/io.js

```diff
@@ -0,0 +1,55 @@
+const fs = require('fs');
+
+module.exports.readFromFileOrStdin = function (path) {
+  const readStream = path ? fs.createReadStream(path) : process.stdin;
+  let inputData = "";
+
+  return new Promise((resolve, reject) => {
+    readStream
+      .setEncoding('utf8')
+      .on('readable', () => {
+        const chunk = readStream.read();
+        if (chunk !== null) {
+          inputData += chunk;
+        }
+      })
+      .on('error', (error) => {
+        reject(error);
+      })
```

Comment:
I'd have
```js
.on('error', reject)
```
but understand if you prefer the parallel construction you have here.